### ISSUE TITLE
MSVC dll version mismatch

### ISSUE THREAD

**Issue by Sharlikran** [loot/loot#665]
Error:

Traceback (most recent call last):
  File "Wrye Bash Launcher.pyw", line 88, in <module>
  File "bash\bash.pyo", line 364, in main
  File "zipextimporter.pyo", line 82, in load_module
  File "bash\bosh\__init__.pyo", line 51, in <module>
  File "zipextimporter.pyo", line 82, in load_module
  File "bash\bosh\mods_metadata.pyo", line 38, in <module>
ImportError: DLL load failed: Impossibile trovare il modulo specificato.

LOOT API used is in [dev-skyrim-patcher-loot-update](https://github.com/Sharlikran/wrye-bash/tree/dev-skyrim-patcher-loot-update).

MSVC Redist installed ```http://download.microsoft.com/download/9/a/2/9a2a7e36-a8af-46c0-8a78-a5eb111eefe2/vc_redist.x86.exe```

LOOT from loot_0.9.2-98-g2f00f5e_dev.7z is working fine.

User has Italian version of Windows 10.

 on my fork. MSVC is at link ```https://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x86.exe```

**Comment by WrinklyNinja** [comment]
That's not the link I give in the readme or installer, you've installed an older version, try [this redistributable](http://download.microsoft.com/download/9/a/2/9a2a7e36-a8af-46c0-8a78-a5eb111eefe2/vc_redist.x86.exe).

**Comment by Sharlikran** [comment]
Still the same error but I am asking for Windows version and language. Is there a way to generate a report similar to dxdiag that would list details about the dll file names and versions he has installed?

**Comment by Sharlikran** [comment]
Language Italian , windows 10. Verified link you provided for msvc did install, no errors.

**Comment by Sharlikran** [comment]
I looked at the dxdiag.txt and it doesn't seem to list dll files and versions.

**Comment by WrinklyNinja** [comment]
Can you run the latest LOOT snapshot? If so, it's not a MSVC redist issue, because it requires the same version of the API.

**Comment by Sharlikran** [comment]
I first thought oh I'm an idiot I forgot to check your documentation.  After he used that link though and had the same error I hoped it was an actual bug.  I can have the user try it.  It's not on my machine.  I'll see what he says.

**Comment by WrinklyNinja** [comment]
Oh, I thought you were experiencing the error.

**Comment by Sharlikran** [comment]
No, it's another user on an Italian version of windows.  My previous comment has a DxDiag.txt but I don't know if that's really useful to you or not.

Wouldn't a Microsoft vc_redist.x86.exe in a foreign language still install msvcr179.dll (I just made that up obviously) just in Italian?

I like your idea of trying LOOT though.  If that fails then getting the exact name of the dll the API wants would be helpful.  I want him to go to his windows folder and manually look for it.

**Comment by Sharlikran** [comment]
Is  loot_0.9.2-98-g2f00f5e_dev.7z good enough for him to test with?

**Comment by WrinklyNinja** [comment]
I said the latest, I don't know when exactly the runtime change happened and I can't check now.

**Comment by Sharlikran** [comment]
Okay thanks.

**Comment by Sharlikran** [comment]
He responded and unfortunately LOOT works fine so now I don't know what to do.  I thought since the error was in the mods_metadata.py file that it might be the loot api Wrye is using.  I guess either I was wrong, I'm just not asking him the right questions, or not providing you the right info.

When you have time, because I figure your busy, if you can think of anything else I can ask him I'd be happy to.

**Comment by WrinklyNinja** [comment]
If LOOT itself works fine, the user has the right runtime installed. How did you get from

```
Traceback (most recent call last):
File "Wrye Bash Launcher.pyw", line 88, in 
File "bash\bash.pyo", line 364, in main
File "zipextimporter.pyo", line 82, in load_module
File "bash\bosh__init__.pyo", line 51, in 
File "zipextimporter.pyo", line 82, in load_module
File "bash\bosh\mods_metadata.pyo", line 38, in 
ImportError: DLL load failed: Impossibile trovare il modulo specificato.
```

to thinking that the MSVC runtime was the problem? Also, where did the error 

```
Error 0x80070666: Cannot install a product when a newer version is installed
```

come from?

**Comment by Sharlikran** [comment]
I assumed the MSVC was the issue when you told me I had not checked your documentation and had suggested an older vc_redist.x86.exe installer.  The error came from the older vc_redist.x86.exe.

With your last two questions, and the fact he is using the standalone EXE I need to find out more information before I go any further.  I want to be helpful not frustrating.  Let me find something out first, let me get back to you.

**Comment by WrinklyNinja** [comment]
Closing this as invalid because all signs point to it being a Wrye Bash issue, probably one with your fork, not a LOOT issue.

**Comment by Sharlikran** [comment]
I don't know all the Git jargon but, if I have this right it was downstream?  Originally I had your loot api changes rebased off of commit https://github.com/wrye-bash/wrye-bash/commit/bc01153bd11ee1a8f15a9ca5ce70db14e3683d34 according to GitHub made 13 days ago.  I use only Python code and never saw any issues nor did Arthmoor.

Upon making a standalone EXE I had this error.
![wrye error_loot](https://cloud.githubusercontent.com/assets/1912426/19432517/b6efef04-9411-11e6-9d55-1d339a347cc3.jpg) which occurred even for Skyrim.

As of commit https://github.com/wrye-bash/wrye-bash/commit/f8bfdc4c5894ca2832aa5aef6515d70f52df110f in the dev branch the error is not present.  I will be rebasing all my branches later today.  I'll try to remember to check that first and check your documentation.  Doing those two things first would have been better.  I still have a lot to learn.

Thanks for your patience.

**Comment by WrinklyNinja** [comment]
Oh, it being a standalone exe build might be the reason, maybe the exe built requires a different resist version and they don't play well together for some reason?

What is used to do the standalone exe build?

**Comment by Sharlikran** [comment]
> maybe the exe built requires a different resist version and they don't play well together for some reason?
> What is used to do the standalone exe build?

I'm not programming savvy enough to know what a resist version is.  

For the wrye-bash repo commit https://github.com/wrye-bash/wrye-bash/commit/f8bfdc4c5894ca2832aa5aef6515d70f52df110f is the most current code.  Your loot api changes are based off of that as of about 9 hours ago.  I believe the code on your fork as of commit https://github.com/WrinklyNinja/wrye-bash/commit/c392e0282e40e61b0fa895f67c3b9d9a9622c3ea will work perfectly as a stand alone EXE.  After I test a new wrye EXE on my machine, I'll see if it's work for the user reporting the error on the closed Beth forum.  I suspect there will be no errors.

Suffice it to say your loot api is working from what I can tell.  I just need to rebase all my branches.

**Comment by Sharlikran** [comment]
> What is used to do the standalone exe build?

I was using https://github.com/Sharlikran/wrye-bash/commit/8266c57c1ce043edabb3f04db03fd89babc7150f to make the standalone EXE last night.

Suffice it to say your loot api changes are working from what I can tell.  I just need to rebase all my branches.

**Comment by WrinklyNinja** [comment]
> I'm not programming savvy enough to know what a "resist version" is.

It's a typo :P I meant redist version. So the exe built from WrinklyNinja/wrye-bash@c392e02 works?

**Comment by Sharlikran** [comment]
> It's a typo :P I meant redist version

Well I didn't know, it could have been something I never heard of.  :smile: 

I just made a standalone EXE and it works on my machine from https://github.com/WrinklyNinja/wrye-bash/commit/c392e0282e40e61b0fa895f67c3b9d9a9622c3ea.  I could not get that to happen list night.  

I'll have the user test it as well and get back to me when he has time, considering time zone difference.  I don't foresee any issues though.

**Comment by Sharlikran** [comment]
I was testing regular Skyrim with your branch at commit https://github.com/WrinklyNinja/wrye-bash/commit/c392e0282e40e61b0fa895f67c3b9d9a9622c3ea but I don't understand something.  If I build the Wrye files I get an EXE installer, a Python version in a 7zip file, and and Standalone EXE in a 7zip file.

If I run the Standalone EXE this is my BashBugDump.log.
```
Wrye Bash starting
Using Wrye Bash Version 307
Python version: 2.7.8
wxPython version: 2.8.12.1 (msw-unicode)
input encoding: None; output encoding: None; locale: ('en_US', 'cp1252')
filesystem encoding: mbcs 
Searching for game to manage:
bush.pyo   71 _supportedGames: Detected the following supported games via Windows Registry:
bush.pyo   73 _supportedGames:  oblivion: G:\Games\steamapps\common\Oblivion
bush.pyo   73 _supportedGames:  skyrim: F:\Games\steamapps\common\Skyrim
bush.pyo   73 _supportedGames:  fallout4: C:\Games\steamapps\common\Fallout 4
bush.pyo  131 _detectGames: Detecting games via the -o argument, bash.ini and relative path:
bush.pyo  137 _detectGames: Set game mode to skyrim found in parent directory of Mopy:  F:\Games\SteamApps\common\Skyrim
bush.pyo  172 setGame: No preferred game specified.
bush.pyo  152 __setGame:  Using skyrim game: F:\Games\SteamApps\common\Skyrim
mods_metadata.pyo   41 <module>: Failed to import the loot_api module.
testing UAC
__init__.pyo 7582 initDirs: Using libbsa API version: 2.0.0
```
There is no odd error with the EXE but the log seems to indicate that it can't access the loot api.

This is what package_for_release.py generated for the standalone.  I made it a zip file since Git doesn't like 7zip files.

[Wrye Bash 307 - Standalone Executable.zip](https://github.com/loot/loot/files/534513/Wrye.Bash.307.-.Standalone.Executable.zip)

The loot api binaries are not in the compiled folder.  If I copy loot_api.dll and loot_api.pyd from an old commit not on your branch, then I get the error.
![wrye error_loot](https://cloud.githubusercontent.com/assets/1912426/19432517/b6efef04-9411-11e6-9d55-1d339a347cc3.jpg)

When someone doesn't use the EXE installer and instead extracts the files from the zip, how do they get the loot binaries?  Why would manually copying them cause that error?

**Comment by WrinklyNinja** [comment]
The binaries are not included as discussed in wrye-bash/wrye-bash#329. If running the install script so that the binaries are installed causes the standalone exe to then error, it's an issue that should be discussed there, not here. It's not a LOOT issue.
